chef/automate
# Node check-in time series API

### :nut_and_bolt: Description: What code changed, and why?

Create an endpoint that creates a time series of daily counts of the number of nodes that have checked in. This has follow on work that also counts the total number of nodes for the daily time series. 

### :chains: Related Resources

https://github.com/chef/automate/issues/2995

### :athletic_shoe: How to Build and Test the Change
1. `build components/config-mgmt-service && build components/automate-gateway && start_all_services`
1. Adding some infra nodes with `chef_load_nodes 100`
1. Run this command to get the time series. `curl -s -f --insecure -H "api-token: $(get_admin_token)" https://localhost/api/v0/cfgmgmt/stats/checkin_counts_timeseries?days_ago=3 | jq`
1. Ensure a time series of the past three days is returned. The time series should end the start of the next hour from now. 

### :white_check_mark: Checklist
